### ISSUE TITLE
Tflite for micro kernel does not have conv2d?

### ISSUE THREAD

**Issue by TF-Deve** [tensorflow/tensorflow#23608]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue  happens on mobile device: No.
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): from master
- Python version: 3.5.2
- Bazel version (if compiling from source): 0.15.0
- GCC/Compiler version (if compiling from source): 4.8
- CUDA/cuDNN version: on CPU
- GPU model and memory: on CPU

**Describe the current behavior**
I'm trying to figure out how to use Tflite for microcontroller, I know it's relatively new to the users. I notice that in the kernel folder, there is only the depth wise conv2d kernel. Does this mean models defined with tf.nn.conv2d won't fit into this framework? Is this part currently under developing or I just simply missed something? Thanks in advance.

**Comment by TF-Deve** [comment]
So far, I've gone through the micro_speech example. I know the model is defined with tf.nn.conv2d. [model defined in this file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/speech_commands/models.py). 
I'm curious about how does tflite for micro run through this model even without a conv2d kernel defined?

**Comment by TF-Deve** [comment]
I'm actually working on my own network. It's a CNN with RGB images as input. It contains regular conv2d layers just like most CNNs do. I wish to know how can I add this kernel into the micro interpreter. Would appreciate it if someone could point me to the correct way. Thanks!

**Comment by TF-Deve** [comment]
@petewarden Could you please give some idea on this? Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by petewarden** [comment]
Sorry for the slow response! We are working on adding more operations (like Conv2D), and hope to have them in soon, but we'd also welcome contributions. I'll be working on some documentation around porting ops, but that isn't ready yet either.

**Comment by TF-Deve** [comment]
If you guys can have a quick guide of how to do it, I can start to add my own kernels. That would be great! Thanks for your reply!

**Comment by bsatrom** [comment]
I'd be interested in this functionality as well. Any updates @petewarden, or is there a guide for contributing kernels, per @TF-Deve's question?
